web/src/main/java/se/inera/intyg/minaintyg/certificate/service/GetCertificateFilterService.java

```diff
@@ -0,0 +1,42 @@
+package se.inera.intyg.minaintyg.certificate.service;
+
+import java.util.List;
+import java.util.function.Function;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.minaintyg.certificate.service.dto.GetCertificateFilterResponse;
+import se.inera.intyg.minaintyg.certificate.service.dto.ListCertificatesRequest;
+import se.inera.intyg.minaintyg.certificate.service.dto.ListCertificatesResponse;
+import se.inera.intyg.minaintyg.integration.api.certificate.model.Certificate;
+import se.inera.intyg.minaintyg.integration.api.certificate.model.CertificateStatusType;
+
+@Service
+@RequiredArgsConstructor
+public class GetCertificateFilterService {
```

Comment:
Nice use of generics!

It takes a little time to understand what is going on in the class. See if you can just restructure the code a little bit for readability. Ideas:
* Move the private method to the end of the class. Starting with the service methods makes it easier to know where to start read the code.
* When building the response consider putting getList(...) on separate line. 
* Putting end parentheses so it matches the start parentheses. If start is on the same line put the end on same line.

Reply:
Agree. Just a note for the future. The new checkstyle moves static methods first and IntelliJ adds static when possible when refactoring. Maybe we can make an update there.